ts/components/RemindEmailValidationOverlay.tsx

```diff
@@ -0,0 +1,141 @@
+/**
+ * A component to remind the user to validate his/her email
+ */
+import I18n from "i18n-js";
+import * as pot from "italia-ts-commons/lib/pot";
+import { untag } from "italia-ts-commons/lib/types";
+import { Button, Content, H2, Text, View } from "native-base";
+import * as React from "react";
+import { Alert, Image } from "react-native";
+import { connect } from "react-redux";
+import { isEmailEditingAndValidationEnabled } from "../config";
+import {
+  navigateBack,
+  navigateToEmailInsertScreen
+} from "../store/actions/navigation";
+import { Dispatch } from "../store/actions/types";
+import { GlobalState } from "../store/reducers/types";
+import TopScreenComponent from "./screens/TopScreenComponent";
+import FooterWithButtons from "./ui/FooterWithButtons";
+
+// tslint:disable-next-line:no-use-before-declare
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps>;
+
+type State = {
+  dispatched: boolean;
+};
+
+class RemindEmailValidationOverlay extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      dispatched: false
+    };
+  }
+  public componentDidUpdate(prevprops: Props, prevstate: State) {
+    const { isValidEmail } = this.props;
+    const { dispatched } = this.state;
+    if (
+      !prevprops.isValidEmail &&
```

Comment:
Being code useful to test the functionality until the overall functionality is completed and the validation check can't be performed, can you add few comments to explain/remember it?

Reply:
@morrys please add a comment describing what is happening